src/output/workspace-impl.cpp

```diff
@@ -398,6 +398,48 @@ class output_viewport_manager_t
         current_vx = 0;
         current_vy = 0;
     }
+    /**
+     * @param threshold Threshold of the view to be counted 
+     *        on that workspace. 1.0 for 100% visible, 0.1 for 10%
+     *
+     * @return a vector of all the workspaces
+     */
+    std::vector<wf::point_t> get_view_workspaces(wayfire_view view, double threshold)
+    {
+        wf::dimensions_t output_workspaces;
+        std::vector<wf::point_t> view_workspaces;
+        wf::output_t *output = view->get_output();
+        output_workspaces = output->workspace->get_workspace_grid_size();
+
+        wlr_box bounding_box;
+        wf::geometry_t output_relative_geometry;
+        wf::geometry_t intersection;
+        double area;
+
+        for (int row = 0; row < output_workspaces.width; row++)
+        {
+           for (int column = 0; column < output_workspaces.heigh; column++)
+           {
+              wf::point_t workspace = {row, column};
+              if (output->workspace->view_visible_on(view, workspace))
+              {
+                 bounding_box = view->get_bounding_box();
+                 output_relative_geometry = view->get_output()->get_relative_geometry();
```

Comment:
The box for a workspace has x/y coordinates equal to vx*output_width / vy*output_height, where vx/vy are the coordinates of a workspace RELATIVE to the current workspace.

FWIW there is render_manager::get_ws_box() which should actually be moved to workspace-manager because that's where it belongs ...